### ISSUE TITLE
Images not rendering after updating to 2.4.0

### ISSUE THREAD

**Issue by anass-naoushi** [Baseflow/flutter_cached_network_image#514]
### Old (and correct) behavior
        no issues.
### Current behavior
       images are not being rendered or displayed.
'package:flutter/src/rendering/object.dart': Failed assertion: line 1273 pos 12: '_debugCanPerformMutations': is not true.

### Reproduction steps
        In normal usages and functions.
### Configuration

**Version:** 2.4.0

**Platform:** 
- iphone: iOS
- robot: Android

**Comment by shunMB** [comment]
Same problem here. We need a solution
My `flutter doctor -v` is 
```
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.6 19G2021, locale ja-JP)    • Flutter version 1.22.3 at /usr/local/bin/flutter
    • Framework revision 8874f21e79 (5 weeks ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/shunmanabe/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.9.3

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • iPhone 12 (mobile) • 61F9D78B-1AFD-45AD-ACE4-10F3B2415363 • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-14-1 (simulator)
```

**Comment by DominicOrga** [comment]
I can also confirm this issue. The following exception is thrown:

══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (20135): The following message was thrown during paint():
I/flutter (20135): Object has been disposed.
I/flutter (20135): 
I/flutter (20135): The relevant error-causing widget was:
I/flutter (20135):   Container
I/flutter (20135):   file:///.../widgets/avatar_frame.dart:90:24
I/flutter (20135): 
I/flutter (20135): When the exception was thrown, this was the stack:
I/flutter (20135): #0      Image.width (dart:ui/painting.dart:1586:37)
I/flutter (20135): #1      paintImage (package:flutter/src/painting/decoration_image.dart:435:31)
I/flutter (20135): #2      DecorationImagePainter.paint (package:flutter/src/painting/decoration_image.dart:276:5)
I/flutter (20135): #3      _BoxDecorationPainter._paintBackgroundImage (package:flutter/src/painting/box_decoration.dart:466:20)
I/flutter (20135): #4      _BoxDecorationPainter.paint (package:flutter/src/painting/box_decoration.dart:484:5)
I/flutter (20135): #5      RenderDecoratedBox.paint (package:flutter/src/rendering/proxy_box.dart:2083:17)
I/flutter (20135): #6      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #7      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #8      RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:131:15)
I/flutter (20135): #9      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #10     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #11     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:131:15)
I/flutter (20135): #12     PaintingContext.pushClipPath.<anonymous closure> (package:flutter/src/rendering/object.dart:527:81)
I/flutter (20135): #13     ClipContext._clipAndPaint (package:flutter/src/painting/clip.dart:30:12)
I/flutter (20135): #14     ClipContext.clipPathAndPaint (package:flutter/src/painting/clip.dart:42:5)
I/flutter (20135): #15     PaintingContext.pushClipPath (package:flutter/src/rendering/object.dart:527:7)
I/flutter (20135): #16     RenderClipOval.paint (package:flutter/src/rendering/proxy_box.dart:1557:23)
I/flutter (20135): #17     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #18     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #19     RenderBoxContainerDefaultsMixin.defaultPaint (package:flutter/src/rendering/box.dart:2616:15)
I/flutter (20135): #20     RenderStack.paintStack (package:flutter/src/rendering/stack.dart:606:5)
I/flutter (20135): #21     RenderStack.paint (package:flutter/src/rendering/stack.dart:614:7)
I/flutter (20135): #22     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #23     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #24     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:131:15)
I/flutter (20135): #25     PaintingContext.pushLayer (package:flutter/src/rendering/object.dart:396:12)
I/flutter (20135): #26     PaintingContext.pushClipPath (package:flutter/src/rendering/object.dart:524:7)
I/flutter (20135): #27     RenderClipOval.paint (package:flutter/src/rendering/proxy_box.dart:1557:23)
I/flutter (20135): #28     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #29     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #30     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:131:15)
I/flutter (20135): #31     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #32     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #33     RenderBoxContainerDefaultsMixin.defaultPaint (package:flutter/src/rendering/box.dart:2616:15)
I/flutter (20135): #34     RenderFlex.paint (package:flutter/src/rendering/flex.dart:971:7)
I/flutter (20135): #35     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #36     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #37     RenderBoxContainerDefaultsMixin.defaultPaint (package:flutter/src/rendering/box.dart:2616:15)
I/flutter (20135): #38     RenderFlex.paint (package:flutter/src/rendering/flex.dart:971:7)
I/flutter (20135): #39     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #40     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #41     RenderShiftedBox.paint (package:flutter/src/rendering/shifted_box.dart:70:15)
I/flutter (20135): #42     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #43     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #44     RenderShiftedBox.paint (package:flutter/src/rendering/shifted_box.dart:70:15)
I/flutter (20135): #45     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #46     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:13)
I/flutter (20135): #47     _RenderSingleChildViewport.paint.paintContents (package:flutter/src/widgets/single_child_scroll_view.dart:588:17)
I/flutter (20135): #48     PaintingContext.pushLayer (package:flutter/src/rendering/object.dart:396:12)
I/flutter (20135): #49     PaintingContext.pushClipRect (package:flutter/src/rendering/object.dart:452:7)
I/flutter (20135): #50     _RenderSingleChildViewport.paint (package:flutter/src/widgets/single_child_scroll_view.dart:592:17)
I/flutter (20135): #51     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2311:7)
I/flutter (20135): #52     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:140:11)
I/flutter (20135): #53     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:100:5)
I/flutter (20135): #54     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:978:29)
I/flutter (20135): #55     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:438:19)
I/flutter (20135): #56     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:914:13)
I/flutter (20135): #57     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:302:5)
I/flutter (20135): #58     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
I/flutter (20135): #59     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
I/flutter (20135): #60     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:864:7)
I/flutter (20135): (elided 4 frames from class _RawReceivePortImpl, class _Timer, and dart:async-patch)
I/flutter (20135): 
I/flutter (20135): The following RenderObject was being processed when the exception was fired: RenderDecoratedBox#03cdb:
Reloaded 0 of 1425 libraries in 308ms.
I/flutter (20135):   creator: DecoratedBox ← ConstrainedBox ← Container ← ClipOval ← Stack ← ClipOval ← Image ← SizedBox
I/flutter (20135):     ← OctoImage ← CachedNetworkImage ← HttpImage ← AvatarFrame ← ⋯
I/flutter (20135):   parentData: <none> (can use size)
I/flutter (20135):   constraints: BoxConstraints(w=170.0, h=170.0)
I/flutter (20135):   size: Size(170.0, 170.0)
I/flutter (20135):   decoration: BoxDecoration:
I/flutter (20135):     image:
I/flutter (20135):       DecorationImage(CachedNetworkImageProvider("...",
I/flutter (20135):       scale: 1.0), center, scale: 1.0)
I/flutter (20135):   configuration: ImageConfiguration(bundle: PlatformAssetBundle#c7f8f(), devicePixelRatio: 2.8,
I/flutter (20135):     locale: en_US, textDirection: TextDirection.ltr, platform: android)
I/flutter (20135): This RenderObject has no descendants.
I/flutter (20135): ════════════════════════════════════════════════════════════════════════════════════════════════════

**Comment by antonvinceguinto** [comment]
My workaround with this was go back to 2.3.3 and remove the ^ symbol in pubspec yaml

**Comment by abdussamad0** [comment]
Same issue

**Comment by renefloor** [comment]
I suspect this is a duplicate of #513 and just fixed in 2.4.1. If it issue persists please reopen the ticket.
